enhancements/cluster-logging/cluster-logging-es-pipeline-processing.md

```diff
@@ -0,0 +1,197 @@
+
+
+### Goals
+
+The high-level goal of this proposal is to lay the groundwork to enable approaches B) and C) as discussed in [LOG-842](https://issues.redhat.com/browse/LOG-842).
+
+Quick comparison of approaches B) and C):
+
+- Approach B) Whenever user needs to store data with a new data model we solve it by pointing this data to new index (or to existing index if there is already one with exactly the same data model). This is very comfy approach for the user except it can lead to very high number of indices and querying experience can be complicated because there are very little or no common conventions of document fields meaning and types.
+
+- Approach C) is on the other side of the spectrum. All data is stored into common index and whenever user needs to store data with a new data model then it is necessary to transfor the data to fit existing data model in existing index. This approach requires attention to data model and users may need to invest time and resources to transform new data sources. On the other hand there is minimum additional indices created in ES cluster and data can be queried uniformly because document fields have predictable meaning and types.
+
+It should be pointed out that Elasticsearch pipeline processes can support both cases. Depending on how much users are willing to pay attention to their data model they can start with approach B) and ventually transition to approach C) once they feel more confident about managing their own data model across more log resources.
+
+In this proposal we assume that the approach B is the one we would start enabling first. But nothing is stopping us from enabling the approach C in the future. Though it is fair to say that approach C will be more complicated to implement because it will require management of custom processing pipelines.
+
+We will be successful when:
+
+* An OCP user (or admin only?)  is able to configure specific logs to be pointed into different (non-default) index in Elasticsearch based on the data contained in the document itself.
+* An OCP user (or admin only?) can provide custom index mapping for the data. This is specifically needed if the [default Elasticsearch type inference mechanism](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-field-mapping.html) does not lead to acceptable results.
+* An OCP user (or admin only?) has enough information and tooling to undertstand and evaluate the impact of creating additional indices. This specifically means new index level metrics will need to be introduced and relevant alerts must be provided as well. Notice that we have recently turned off index level Prometheus metrics (because they were high-cardinality metrics) thus we will need to revisit index level metrics and provide new aggregated non-high cardinality metrics.
```

Comment:
This is a big one imho. Thanks for putting that in.

Reply:
np, yw :-)